File: src/OnPeople/BackEnd/src/OnPeople.Domain/Models/Funcionarios/Funcionario.cs
using OnPeople.Domain.Models.Cargos;
using OnPeople.Domain.Models.Departamentos;
using OnPeople.Domain.Models.Empresas;
using OnPeople.Domain.Models.Users;

namespace OnPeople.Domain.Models.Funcionarios
{
    public class Funcionario
    {
        public int Id { get; set; }
        public  string NomeCompleto{ get; set; }
        public string DataAdmissao { get; set; }
        public string DataDemissao { get; set; }
        public Boolean Ativo { get; set; }
        public int UserId { get; set; }
        public User Users { get; set; }
        public int DepartamentoId { get; set; }
        public Departamento Departamento { get; set; }
        public int CargoId { get; set; }
        public Cargo Cargo { get; set; }
        public int EmpresaId { get; set; }
        public Empresa Empresa { get; set; }
        public int Funcao { get; set; }
        public IEnumerable<DadoPessoal> DadosPessoais { get; set; }
        public IEnumerable<Endereco> Enderecos { get; set; }
        public IEnumerable<FuncionarioMeta> FuncionariosMetas { get; set; }
    }
}

File: src/OnPeople/FrontEnd/OnPeople-App/src/app/employees/models/Funcionario.ts
import { Empresa } from "src/app/companies/models";
import { Departamento } from "src/app/departments/models";
import { Cargo } from "src/app/jobRoles/models";
import { Users } from "src/app/users/models";
import { DadoPessoal } from "./DadoPessoal";
import { Endereco } from "./Endereco";
import { FuncionarioMeta } from "./FuncionarioMeta";

export interface Funcionario {
  id: number;
  nomeCompleto: string;
  dataAdmissao: string;
  dataDemissao: string;
  ativo: Boolean;
  userId: number;
  user: Users;
  departamentoId: number;
  departamento: Departamento;
  cargoId: number;
  cargo: Cargo;
  empresaId: number;
  empresa: Empresa;
  funcao: string;
  dadosPessoais: DadoPessoal[];
  enderecos: Endereco[];
  funcionariosmetas: FuncionarioMeta[];
}
